From 55f801b368b026f680dbb81423f7f88ed0711392 Mon Sep 17 00:00:00 2001
From: Evgeni Golov <evgeni@golov.de>
Date: Sun, 6 Aug 2023 14:41:03 +0200
Subject: [PATCH] don't cast smapi_port in inline assembly

the generated assembly looks identical with and without the cast on
GCC-13

Fixes: #41
---
 tp_smapi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tp_smapi.c b/tp_smapi.c
index de7391e..5be638b 100644
--- a/tp_smapi.c
+++ b/tp_smapi.c
@@ -201,7 +201,7 @@ static int smapi_request(u32 inEBX, u32 inECX,
 			 "=m"(tmpEDI),
 			 "=m"(tmpESI)
 			:"m"(inEBX), "m"(inECX), "m"(inEDI), "m"(inESI),
-			 "m"((u16)smapi_port)
+			 "m"(smapi_port)
 			:"%eax", "%ebx", "%ecx", "%edx", "%edi",
 			 "%esi");
 
